### PR TITLE
Small grammatical and stylistic fixes

### PR DESCRIPTION
**Pull request by anoadragon453**
Read through the S2S doc and noticed a lot of little things that needed to be fixed. This PR is the culmination of those fixes.

### CODE REVIEW

File: specification/server_server_api.rst [hunk 1]
```diff
@@ -500,119 +500,121 @@ the state of the room.
           state of the room. For example, a redacted ``join`` event will still
           result in the user being considered joined.
 
+The rules are as follows:
+
 1. If type is ``m.room.create``, allow if and only if it has no
    previous events - *i.e.* it is the first event in the room.
 
-#. If type is ``m.room.member``:
```
Comment (anoadragon453) [inline specification/server_server_api.rst]: The extra space in each indentation caused the documentation parser to not indent anything. This fixes that.

Additionally, while RST can auto-enumerate the proceeding letters and numbers by simply putting a `#` after `a.` or `i.`, this did not work if another list was in the middle of this enumeration, causing many little errors. To fix this, only the leaf lists are now auto-enumerated.

File: scripts/css/blockquote.css [hunk 2]
```diff
@@ -1,5 +1,4 @@
 blockquote {
     margin: 20px 0 30px;
-    border-left: 5px solid;
```
Comment (anoadragon453) [inline scripts/css/blockquote.css]: This caused an ugly black block to appear next to the rules list when properly enumerated. I *assumed* this wasn't intended behavior. I could not find any other usage of blockquotes in the documentation.

Comment (anoadragon453) [reply scripts/css/blockquote.css]: image

**Comment by ara4n** [comment]
oh, awesome - thank you! i have one minor nit but otherwise lgtm

File: specification/server_server_api.rst [hunk 3]
```diff
@@ -934,20 +943,22 @@ event to an resident homeserver, by using the ``send_join`` endpoint. This is
 invoked using the room ID and the event ID of the new member event.
 
 The resident homeserver then accepts this event into the room's event graph,
-and responds to the joining server with the full set of state for the newly-
+and responds to the joining server with the full set of state for the newly
```
[the remark this answers is not in the document]
Comment (anoadragon453) [reply specification/server_server_api.rst]: Alright, it creates "newly- joined" in the spec, but I'll move "newly-" to the next line to fix that.

**Comment by ara4n** [comment]
woo, thanks

**Pull request merged**